### ISSUE TITLE
Canvas is failing when parameters are of type `np.float64`

### ISSUE THREAD

**Issue by saroad2** [beeware/toga#1559]
**Describe the bug**

After the new changes in version 3.0 of Pythonnet, there is not longer an implicit conversion between `np.float64` to `float`.
This causes crashes when trying to use `Canvas` such as in issue beeware/toga-chart#16.

While some of the problem was fixes in #1558, some of the errors are still occurring. Here is an example I encountered in [Eddington-GUI](https://github.com/EddLabs/eddington-gui):

```
Traceback (most recent call last):
  File "C:\Users\sagi\PycharmProjects\eddington-gui\venv\lib\site-packages\toga_winforms\widgets\canvas.py", line 172, in bezier_curve_to
    PointF(*draw_context.last_point),
  File "C:\Users\sagi\PycharmProjects\eddington-gui\venv\lib\site-packages\toga\widgets\canvas.py", line 876, in _draw
    impl.bezier_curve_to(
  File "C:\Users\sagi\PycharmProjects\eddington-gui\venv\lib\site-packages\toga\widgets\canvas.py", line 43, in _draw
    obj._draw(impl, *args, **kwargs)
  File "C:\Users\sagi\PycharmProjects\eddington-gui\venv\lib\site-packages\toga\widgets\canvas.py", line 407, in _draw
    obj._draw(impl, *args, **kwargs)
  File "C:\Users\sagi\PycharmProjects\eddington-gui\venv\lib\site-packages\toga\widgets\canvas.py", line 43, in _draw
    obj._draw(impl, *args, **kwargs)
  File "C:\Users\sagi\PycharmProjects\eddington-gui\venv\lib\site-packages\toga_winforms\widgets\canvas.py", line 93, in winforms_paint
    self.interface._draw(self, draw_context=context)
   at Python.Runtime.PythonException.ThrowLastAsClrException()
   at Python.Runtime.Dispatcher.TrueDispatch(Object[] args)
   at Python.Runtime.Dispatcher.Dispatch(Object[] args)
   at __System_Windows_Forms_PaintEventHandlerDispatcher.Invoke(Object , PaintEventArgs )
   at System.Windows.Forms.Control.OnPaint(PaintEventArgs e)
   at System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   at System.Windows.Forms.Control.WmPaint(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
No method matches given arguments for PointF..ctor: (<class 'numpy.float64'>, <class 'numpy.float64'>)
```

**To Reproduce**
1. Go to the `Canvas` example in a Windows computer
2. Try to use `np.float64` instead of `float`
3. Run and see it crash

**Expected behavior**
It should not crash.

**Environment:**
 - Operating System: Windows
 - Python version: 3.10
 - Software versions:
   - Briefcase: Latest
   - Toga: Latest
   - ...

**Additional context**

**Comment by saroad2** [comment]
As before, at seems like a quick fix. Just adding more conversions in the Winforms implementation to `float`.
@freakboy3742 , I just need you OK that I can make that change.

**Comment by freakboy3742** [comment]
@saroad2 +1 - I fixed the casts that were obvious in #1558, but if you're seeing additional ones, then go ahead and cast as required.
